File: Components/WinForms/VB/PivotGridMainDemo/ProgressForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms

Namespace DevExpress.XtraPivotGrid.Demos
	Partial Public Class ProgressForm
		Inherits DevExpress.XtraEditors.XtraForm
		Public Sub New(ByVal parent As Form)
			InitializeComponent()
			If parent IsNot Nothing Then
				Left = parent.Left + CInt(Fix((parent.Width - Width) / 2))
				Top = parent.Top + CInt(Fix((parent.Height - Height) / 2))
			End If
			Me.Height = progressBarControl1.Height + progressBarControl1.Top * 2 + 4
		End Sub

		Public Sub SetProgressValue(ByVal position As Integer)
			progressBarControl1.Position = position
			Me.Update()
		End Sub
	End Class
End Namespace

File: Components/WinForms/CS/PivotGridMainDemo/ProgressForm.cs
using System;
using System.Windows.Forms;

namespace DevExpress.XtraPivotGrid.Demos {
	public partial class ProgressForm : DevExpress.XtraEditors.XtraForm {
		public ProgressForm(Form parent) {
			InitializeComponent();
			if(parent != null) {
                Left = parent.Left + (int)((parent.Width - Width) / 2);
                Top = parent.Top + (int)((parent.Height - Height) / 2);
			}
			this.Height = progressBarControl1.Height + progressBarControl1.Top * 2 + 4;
		}

		public void SetProgressValue(int position) {
			progressBarControl1.Position = position;
			this.Update();
		}
	}
}
